BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+
+        /**
+         * Creates a new [Configuration] instance from a json string.
+         *
+         * @param configurationString The json configuration string from Braintree.
+         * @return [Configuration] instance.
+         */
+        @JvmStatic
+        @Throws(JSONException::class)
+        @RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+        fun fromJson(configurationString: String?): Configuration {
+            return Configuration(configurationString)
+        }
+    }
+
+    private val configurationString: String
+    private val challenges: MutableSet<String> = HashSet()
+
+    private val braintreeApiConfiguration: BraintreeApiConfiguration
+    private val analyticsConfiguration: AnalyticsConfiguration
+    private val cardConfiguration: CardConfiguration
+
+    private val payPalConfiguration: PayPalConfiguration
+    private val googlePayConfiguration: GooglePayConfiguration
+
+    private val venmoConfiguration: VenmoConfiguration
+    private val kountConfiguration: KountConfiguration
+    private val unionPayConfiguration: UnionPayConfiguration
+    private val visaCheckoutConfiguration: VisaCheckoutConfiguration
+    private val graphQLConfiguration: GraphQLConfiguration
+    private val samsungPayConfiguration: SamsungPayConfiguration
+
+    /**
+     * @return The assets URL of the current environment.
+     */
+    open val assetsUrl: String
+
+    /**
+     * @return The url of the Braintree client API for the current environment.
+     */
+    open val clientApiUrl: String
+
+    /**
+     * @return The current environment.
+     */
+    open val environment: String
+
+    /**
+     * @return the current Braintree merchant id.
+     */
+    open val merchantId: String
+
+    /**
+     * @return `true` if PayPal is enabled and supported in the current environment,
+     * `false` otherwise.
+     */
+    open val isPayPalEnabled: Boolean
+
+    /**
+     * @return `true` if 3D Secure is enabled and supported for the current merchant account,
+     * `false` otherwise.
+     */
+    open val isThreeDSecureEnabled: Boolean
+
+    /**
+     * @return the current Braintree merchant account id.
+     */
+    open val merchantAccountId: String?
+
+    /**
+     * @return the JWT for Cardinal
+     */
+    open val cardinalAuthenticationJwt: String?
+
+    init {
+        if (configurationString == null) {
+            throw JSONException("Configuration cannot be null")
+        }
+
+        this.configurationString = configurationString
+        val json = JSONObject(configurationString)
+        assetsUrl = Json.optString(json, ASSETS_URL_KEY, "")
+        clientApiUrl = json.getString(CLIENT_API_URL_KEY)
+
+        // parse json challenges
+        json.optJSONArray(CHALLENGES_KEY)?.let { challengesArray ->
+            for (i in 0 until challengesArray.length()) {
+                challenges.add(challengesArray.optString(i, ""))
+            }
```

Comment:
It turns out we need to keep this because `JSONArray` doesn't have the same interface as a standard Kotlin array. It's awesome that we have a default JSON parser that ships with Android, but its interface is a bit dated.

In the future, we may add some extensions to it like we did in Northstar to make it easier to work with.